core/src/main/scala/libretto/scaletto/ScalettoLib.scala

```diff
@@ -54,49 +54,46 @@ class ScalettoLib[
   def const[A](a: A): One -⚬ Val[A] =
     andThen(done, constVal(a))
 
-  implicit def junctionVal[A]: Junction.Positive[Val[A]] =
+  given junctionVal[A]: Junction.Positive[Val[A]] =
     new Junction.Positive[Val[A]] {
       override def awaitPosFst: (Done |*| Val[A]) -⚬ Val[A] =
         par(constVal(()), id[Val[A]]) > unliftPair > mapVal(_._2)
     }
 
-  implicit def junctionNeg[A]: Junction.Negative[Neg[A]] =
-    new Junction.Negative[Neg[A]] {
-      override def awaitNegFst: Neg[A] -⚬ (Need |*| Neg[A]) =
-        contramapNeg[(Unit, A), A](_._2) > liftNegPair > par(constNeg(()), id[Neg[A]])
-    }
+  given junctionNeg[A]: Junction.Negative[Neg[A]] with {
+    override def awaitNegFst: Neg[A] -⚬ (Need |*| Neg[A]) =
+      contramapNeg[(Unit, A), A](_._2) > liftNegPair > par(constNeg(()), id[Neg[A]])
+  }
 
-  implicit def signalingVal[A]: Signaling.Positive[Val[A]] =
+  given signalingVal[A]: Signaling.Positive[Val[A]] =
     new Signaling.Positive[Val[A]] {
       override def notifyPosFst: Val[A] -⚬ (Ping |*| Val[A]) =
         notifyVal
     }
 
-  implicit def signalingNeg[A]: Signaling.Negative[Neg[A]] =
-    new Signaling.Negative[Neg[A]] {
-      override def notifyNegFst: (Pong |*| Neg[A]) -⚬ Neg[A] =
-        notifyNeg
-    }
+  given signalingNeg[A]: Signaling.Negative[Neg[A]] with {
+    override def notifyNegFst: (Pong |*| Neg[A]) -⚬ Neg[A] =
+      notifyNeg
+  }
 
-  implicit def signalingJunctionPositiveVal[A]: SignalingJunction.Positive[Val[A]] =
+  given signalingJunctionPositiveVal[A]: SignalingJunction.Positive[Val[A]] =
     SignalingJunction.Positive.from(
       signalingVal,
       junctionVal,
     )
 
-  implicit def signalingJunctionNegativeNeg[A]: SignalingJunction.Negative[Neg[A]] =
+  given signalingJunctionNegativeNeg[A]: SignalingJunction.Negative[Neg[A]] =
     SignalingJunction.Negative.from(
-      signalingNeg[A],
-      junctionNeg[A],
+      signalingNeg,
+      junctionNeg,
     )
 
-  implicit def valNegDuality[A]: Dual[Val[A], Neg[A]] =
-    new Dual[Val[A], Neg[A]] {
-      val lInvert: One -⚬ (Neg[A] |*| Val[A]) = promise[A]
-      val rInvert: (Val[A] |*| Neg[A]) -⚬ One = fulfill[A]
-    }
+  given valNegDuality[A]: Dual[Val[A], Neg[A]] with {
+    val lInvert: One -⚬ (Neg[A] |*| Val[A]) = promise[A]
+    val rInvert: (Val[A] |*| Neg[A]) -⚬ One = fulfill[A]
+  }
 
-  implicit def negValDuality[A]: Dual[Neg[A], Val[A]] =
+  given negValDuality[A]: Dual[Neg[A], Val[A]] =
```

Comment:
Did I mention that those are used a lot explicitly? :smile:

Reply:
Maybe no `Dual`s should be implicit/given:

You might have `Dual[A, B]`, but (if your DSL is at least `InvertDSL`) you also have `Dual[A, -[A]]`.

And then whenever you have `Dual[A, B]`, you also have `Dual[B, A]`. Sometimes you can say which one is more "basic" and which one is derived, but often not.